### PR TITLE
Serve raw files with the correct content types

### LINKED ISSUE #8152: [API] Provide correct MIME type when getting a raw text file
- Gitea version (or commit ref): 1.10.0+dev-274-g3fd0eec90
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant

## Description

When fetching a raw text file via the API the content type header does not reflect the file's MIME type.
https://try.gitea.io/api/v1/repos/go-gitea/gitea/raw/public/js/draw.js

`Content-Type: text/plain; charset=utf-8`

Expected result would be:

`Content-Type: application/javascript`

When fetching an image file via the API, the content type seems to be set correctly though:
https://try.gitea.io/api/v1/repos/go-gitea/gitea/raw/public/img/404.png

`Content-Type: image/png`

This is related to issue #7620.

### PR DESCRIPTION
**Pull request by tslocum**
Resolves #8152.

This is an alternative implementation to the existing PR for this issue.

### CODE REVIEW

**Comment by silverwind** [comment]
I think this should be optional and default off for the sake of security. See https://github.com/go-gitea/gitea/issues/8152#issuecomment-814351404.

File: routers/repo/download.go [hunk 1]
```diff
@@ -41,25 +41,31 @@ func ServeData(ctx *context.Context, name string, size int64, reader io.Reader)
...
+		if strings.Contains(ct, base.SVGMimeType) {
 			ctx.Resp.Header().Set("Content-Security-Policy", "default-src 'none'; style-src 'unsafe-inline'; sandbox")
 			ctx.Resp.Header().Set("X-Content-Type-Options", "nosniff")
-			ctx.Resp.Header().Set("Content-Type", base.SVGMimeType)
 		}
 	} else {
 		ctx.Resp.Header().Set("Content-Disposition", fmt.Sprintf(`attachment; filename="%s"`, name))
 		ctx.Resp.Header().Set("Access-Control-Expose-Headers", "Content-Disposition")
 	}
+	ctx.Resp.Header().Set("Content-Type", strings.ToLower(ct))
```
Comment (silverwind) [inline routers/repo/download.go:81]: Move `X-Content-Type-Options` here as well to opt-out mime sniffing.

**Comment by tslocum** [comment]
Thanks @silverwind.  PR updated.

**Comment by silverwind** [comment]
See https://github.com/go-gitea/gitea/issues/8152#issuecomment-819724255 for a better suggestion.

**Comment by tslocum** [comment]
While I can see merit in your suggestion, the changes in this PR could be explicitly enabled by users via a configuration option (as you originally suggested), with a warning stating the potential security implications.  It would also coexist with custom content types when they are implemented (i.e., check custom content types before detecting a file's content type).

Closing because the scope of the solution has grown beyond my interest.

**Pull request closed without merging**